docs/tutorial/202-pipeline.md

```diff
@@ -0,0 +1,300 @@
+# Agent Interactions: Dive deeper into Pipelines and Message Hub
+
+**Pipeline & MsgHub** (message hub) are one or a sequence of steps describing how the structured `Msg` passes between multi-agents, which streamlines the process of collaboration across agents.
+
+`Pipeline` allows users to program communication among agents easily, and `MsgHub` enables message sharing among agents like a group chat.
+
+## Pipelines
+
+`Pipeline` in AgentScope serves as conduits through which messages pass among agents. In AgentScope, an `Agent` is a subclass of an `Operator` that performs some operation on input data. Pipelines extend this concept by encapsulating multiple agents, and also act as an `Operator`.
+
+Here is the base class for all pipeline types:
+
+```python
+class PipelineBase(Operator):
+  	"""Base interface of all pipelines."""
+    # ... [code omitted for brevity]
+    @abstractmethod
+    def __call__(self, x: Optional[dict] = None) -> dict:
+        """Define the actions taken by this pipeline.
+
+        Args:
+            x (Optional[`dict`], optional):
+                Dialog history and some environmental information
+
+        Returns:
+            `dict`: The pipeline's response to the input.
+        """
+```
+
+### Category
+
+AgentScope provides two main types of pipelines based on their implementation strategy:
+
+* **Operator-Type Pipelines**
+
+  * These pipelines are object-oriented and inherit from the `PipelineBase`. They are operators themselves and can be combined with other operators to create complex interaction patterns.
+
+    ```python
+    # Instantiate and invoke
+    pipeline = ClsPipeline(agent1, agent2, agent3)
+    x = pipeline(x)
+    ```
+
+* **Functional Pipelines**
+
+  * Functional pipelines provide similar control flow mechanisms as the class-based pipelines but are implemented as standalone functions. These are useful for scenarios where a class-based setup may not be necessary or preferred.
+
+    ```python
+    # Just invoke
+    x = funcpipeline(x, agent1, agent2, agent3)
```

Comment:
`x` should be place in the last, and agents should be contained in a list

@pan-x-c , pls ensure it.